src/commands/miscellaneous/help.ts

```diff
@@ -0,0 +1,155 @@
+ * to access the help embed of each command
+ */
+
+// The default help window to show when no topic is specified
+const defaultEmbed: EmbedBuilder = new EmbedBuilder()
+    .setTitle("General Help")
+    .setDescription("Here are my commands! Use `/help <topic>` to get help for a specific command")
+    .setColor(MAGENTA);
+
+
+// ----- LOAD HELP COMMAND EMBEDS -----
+
+const commands: Map<string, EmbedBuilder> = new Map();
+
+// Get the path of the commands directory (./src) > "commands"
+const cmdFoldersPath: string = path.join(__dirname, "..");
+const cmdFolders: string[] = fs.readdirSync(cmdFoldersPath);
+
+// Look through each command category folder
+for (const folder of cmdFolders) {
+    // Get the path of the current directory (./src) > "commands" > category
+    const cmdFolder: string = path.join(cmdFoldersPath, folder);
+    const cmdFiles: string[] = fs.readdirSync(cmdFolder).filter(f => f.endsWith(".js"));
+
+    let categoryCommands: string = "";
+
+    // Look through each JS file in each command category folder
+    for (const file of cmdFiles) {
+        // Skip this file
+        if (file === "help.js") {
+            categoryCommands += "`/help`, ";
+            continue;
+        }
+        
+        // Get the path of the current directory (./src) > "commands" > category > command
+        const cmdPath: string = path.join(cmdFolder, file);
+        const cmd: Command = require(cmdPath).command;
+        
+        // If the imported file is a valid command, add it
+        if ("data" in cmd && "execute" in cmd && "help" in cmd) {
+            const name: string = cmd.data.name;
+            // The colour is set to magenta here
+            commands.set(name, cmd.help.setColor(MAGENTA));
+            categoryCommands += `\`/${name}\`, `;
+        }
+    }
+
+    // Remove the dangling comma
+    // eslint-disable-next-line no-magic-numbers
+    categoryCommands = categoryCommands.slice(0, categoryCommands.length - 2);
+
+    // Capitalise the category name
+    const name: string = folder.replace(folder[0], folder[0].toUpperCase());
+    defaultEmbed.addFields({ name: name, value: categoryCommands });
+}
+
+const name: string = "help";
+
+// Help command embed
+const help: EmbedBuilder = new EmbedBuilder()
+    .setTitle("Help")
+    .setDescription("The help command which displays useful command information!")
+    .addFields(
+        { name: "Format", value: `\`/${name} [topic]\`` },
+        { name: "[topic]", value: "Optional parameter. The command that you want to know more about" }
+    )
+    .setColor(MAGENTA);
+
+// Add the help field of the help command
+commands.set("help", help);
+
+// The command choices for the help command
+// with the required fields in each choice
+const choices = Array.from(commands.keys()).map((cmd: string): { name: string, value: string } => {
+    return { name: cmd, value: cmd };
+});
+
+// ----- END LOAD HELP COMMAND EMBEDS -----
```

Comment:
The `/help` command is a little different from the other commands because it needs to load the help embed for every other command, almost like what the bot does in `index.ts`